composer/datasets/evaluator.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import List, Optional
+
+import yahp as hp
+
+from composer.core.types import Evaluator
+from composer.datasets import DataloaderHparams
+from composer.datasets.dataset_registry import get_dataset_registry
+from composer.datasets.hparams import DatasetHparams
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluatorHparams(hp.Hparams):
+    """Params for the :class:`Evaluator`.
+
+    See the documentation for the :class:`Evaluator`.
+    """
+    hparams_registry = {  # type: ignore
+        "eval_dataset": get_dataset_registry(),
+    }
+
+    label: str = hp.required(doc="Name of the Evaluator object. Used for logging/reporting metrics")
+    eval_dataset: DatasetHparams = hp.required(doc="Evaluator dataset for the Evaluator")
+    metric_names: Optional[List[str]] = hp.optional(
```

Comment:
Should it be a `List[str]`? Curious when it would be none?

Reply:
Also should it be required?

Reply:
Let's freeze the API design for now. We can make subsequent improvements, but I don't want to delay this PR any further. The optional use case here is if the user has multiple datasets and wants to apply all the metrics provided by the model's `metrics() -> MetricCollection` method.